### PR TITLE
[Streaming]Streaming data transfer support cross language

### PR DESCRIPTION
**Pull request by MissiontoMars**
## Why are these changes needed?
This PR adds cross language in the streaming data transfer layer, to support the following Cross-Lang API([PR](https://github.com/ray-project/ray/pull/7464)).

In Cross-Lang design, the peer's actor type is uncertain, so the entry point function used by direct call should be configurable.

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://ray.readthedocs.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

**Comment by MissiontoMars** [comment]
@chaokunyang

File: streaming/src/queue/queue_handler.cc [hunk 1]
```diff
@@ -83,10 +78,11 @@ std::shared_ptr<Transport> QueueMessageHandler::GetOutTransport(
...
 
 void QueueMessageHandler::SetPeerActorID(const ObjectID &queue_id,
-                                         const ActorID &actor_id) {
+                                         const ActorID &actor_id, RayFunction &async_func,
+                                         RayFunction &sync_func) {
   actors_.emplace(queue_id, actor_id);
-  out_transports_.emplace(queue_id,
-                          std::make_shared<ray::streaming::Transport>(actor_id));
+  out_transports_.emplace(queue_id, std::make_shared<ray::streaming::Transport>(
+                                        actor_id, async_func, sync_func));
```
Comment (chaokunyang) [inline streaming/src/queue/queue_handler.cc:85]: lint

File: streaming/java/streaming-runtime/src/main/java/org/ray/streaming/runtime/transfer/ChannelInitialParameters.java [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+import org.ray.runtime.actor.NativeRayPyActor;
+import org.ray.runtime.functionmanager.FunctionDescriptor;
+import org.ray.runtime.functionmanager.JavaFunctionDescriptor;
+import org.ray.runtime.functionmanager.PyFunctionDescriptor;
+import org.ray.streaming.runtime.worker.JobWorker;
+
+/**
+ * Save channel initial parameters needed by DataWriter/DataReader.
+ */
+public class ChannelInitialParameters {
```
[the remark this answers is not in the document]
Comment (MissiontoMars) [reply streaming/java/streaming-runtime/src/main/java/org/ray/streaming/runtime/transfer/ChannelInitialParameters.java]: `ChannelCreationParameters ` is better.

**Comment by MissiontoMars** [comment]
@jovany-wang

**Review by chaokunyang — APPROVED** [review]
LGTM, thanks

**Review by jovany-wang — APPROVED** [review]

**Pull request merged**